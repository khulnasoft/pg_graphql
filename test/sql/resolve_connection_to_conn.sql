begin;

    create table account(
        id serial primary key,
        email varchar(255) not null
    );


    insert into public.account(email)
    values
        ('aardvark@x.com'),
        ('bat@x.com'),
        ('cat@x.com'),
        ('dog@x.com'),
        ('elephant@x.com');


    create table blog(
        id serial primary key,
        owner_id integer not null references account(id),
        name varchar(255) not null
    );
    comment on table blog is e'@graphql({"totalCount": {"enabled": true}})';


    insert into blog(owner_id, name)
    values
        ((select id from account where email ilike 'a%'), 'A: Blog 1'),
        ((select id from account where email ilike 'a%'), 'A: Blog 2'),
        ((select id from account where email ilike 'a%'), 'A: Blog 3'),
        ((select id from account where email ilike 'b%'), 'B: Blog 4');


    select jsonb_pretty(
        graphql.resolve($$
    {
      accountCollection {
        edges {
          node {
            id
            email
            blogCollection {
              totalCount
                edges {
                  node {
                    name
                }
              }
            }
          }
        }
      }
    }
        $$)
    );


rollback;
