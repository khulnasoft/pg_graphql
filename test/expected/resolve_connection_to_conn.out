begin;
    create table account(
        id serial primary key,
        email varchar(255) not null
    );
    insert into public.account(email)
    values
        ('aardvark@x.com'),
        ('bat@x.com'),
        ('cat@x.com'),
        ('dog@x.com'),
        ('elephant@x.com');
    create table blog(
        id serial primary key,
        owner_id integer not null references account(id),
        name varchar(255) not null
    );
    comment on table blog is e'@graphql({"totalCount": {"enabled": true}})';
    insert into blog(owner_id, name)
    values
        ((select id from account where email ilike 'a%'), 'A: Blog 1'),
        ((select id from account where email ilike 'a%'), 'A: Blog 2'),
        ((select id from account where email ilike 'a%'), 'A: Blog 3'),
        ((select id from account where email ilike 'b%'), 'B: Blog 4');
    select jsonb_pretty(
        graphql.resolve($$
    {
      accountCollection {
        edges {
          node {
            id
            email
            blogCollection {
              totalCount
                edges {
                  node {
                    name
                }
              }
            }
          }
        }
      }
    }
        $$)
    );
                        jsonb_pretty                         
-------------------------------------------------------------
 {                                                          +
     "data": {                                              +
         "accountCollection": {                             +
             "edges": [                                     +
                 {                                          +
                     "node": {                              +
                         "id": 1,                           +
                         "email": "aardvark@x.com",         +
                         "blogCollection": {                +
                             "edges": [                     +
                                 {                          +
                                     "node": {              +
                                         "name": "A: Blog 1"+
                                     }                      +
                                 },                         +
                                 {                          +
                                     "node": {              +
                                         "name": "A: Blog 2"+
                                     }                      +
                                 },                         +
                                 {                          +
                                     "node": {              +
                                         "name": "A: Blog 3"+
                                     }                      +
                                 }                          +
                             ],                             +
                             "totalCount": 3                +
                         }                                  +
                     }                                      +
                 },                                         +
                 {                                          +
                     "node": {                              +
                         "id": 2,                           +
                         "email": "bat@x.com",              +
                         "blogCollection": {                +
                             "edges": [                     +
                                 {                          +
                                     "node": {              +
                                         "name": "B: Blog 4"+
                                     }                      +
                                 }                          +
                             ],                             +
                             "totalCount": 1                +
                         }                                  +
                     }                                      +
                 },                                         +
                 {                                          +
                     "node": {                              +
                         "id": 3,                           +
                         "email": "cat@x.com",              +
                         "blogCollection": {                +
                             "edges": [                     +
                             ],                             +
                             "totalCount": 0                +
                         }                                  +
                     }                                      +
                 },                                         +
                 {                                          +
                     "node": {                              +
                         "id": 4,                           +
                         "email": "dog@x.com",              +
                         "blogCollection": {                +
                             "edges": [                     +
                             ],                             +
                             "totalCount": 0                +
                         }                                  +
                     }                                      +
                 },                                         +
                 {                                          +
                     "node": {                              +
                         "id": 5,                           +
                         "email": "elephant@x.com",         +
                         "blogCollection": {                +
                             "edges": [                     +
                             ],                             +
                             "totalCount": 0                +
                         }                                  +
                     }                                      +
                 }                                          +
             ]                                              +
         }                                                  +
     }                                                      +
 }
(1 row)

rollback;
